# Example 1

Consider the below second stage problem:

1.  find a lower bound $L \leq \min_x \big\{ Q(x) | Ax = b , x \in  X \big\}$

2. derive a cut of the type  $\theta \geq (qs −L)
\Big(
\sum_{i \in S} x_i -
\sum_{i \in S} x_i \Big)
-
- (qs - L)(|S| −1) +L$

$$\min \quad −2y_1 −3y_2$$
$$\text{s. t.} \quad y_1 +2y_2 \leq \xi_1 - x1$$
$$y1 \leq \xi_2 −x2$$
$$y \in Z^+$$

- $\xi = (2,2), p = 0.5$

- $\xi = (4,3), p= 0.5$

- $x = (0,1)$

# Step 1: finding the lower bound $L$

- $\min -2 y_1 -3 y_2$ can be written as $- (\max 2 y_1 + 3 y_2)$ to maximise this the $y$ should be as large as possible

- we can do this by considering $x = (0,0)$ leading to:

$$y_1 +2y_2 \leq \xi_1 $$
$$y1 \leq \xi_2$$

- We drop the integrality requirment before caluculations

- Now this is our problem

$$\min \quad −2y_1 −3y_2$$
$$\text{s. t.} \quad y_1 +2y_2 \leq \xi_1$$
$$y1 \leq \xi_2$$
$$y \in R$$



For - $\xi = (2,2), p = 0.5$ we have:

$$\min \quad −2y_1 −3y_2$$
$$\text{s. t.} \quad y_1 +2y_2 \leq 2$$
$$y1 \leq 2$$
$$y \in R^+$$


In [8]:
using GLPK, JuMP
m = Model(GLPK.Optimizer)
c = [-2 , -3]
A = [1 2 ; 1 0]
b = [2 ; 2]
@variable(m, 0 ≤ y[1:2])
@objective(m, Min, c' * y)
@constraint(m, c1, A * y .≤ b)
optimize!(m)
println("y = ", value.(y) )
println("obj = ", objective_value(m) )

y = [2.0, 0.0]
obj = -4.0


For - $\xi = (4,3), p = 0.5$ we have:

$$\min \quad −2y_1 −3y_2$$
$$\text{s. t.} \quad y_1 +2y_2 \leq 4$$
$$y1 \leq 3$$
$$y \in R^+$$


In [9]:
using GLPK, JuMP
m = Model(GLPK.Optimizer)
c = [-2 , -3]
A = [1 2 ; 1 0]
b = [4 ; 3]
@variable(m, 0 ≤ y[1:2])
@objective(m, Min, c' * y)
@constraint(m, c1, A * y .≤ b)
optimize!(m)
println("y = ", value.(y) )
println("obj = ", objective_value(m) )

y = [3.0, 0.5]
obj = -7.5


In [11]:
(-7.5 -4)/2

-5.75

Therefore, the lower bound is $L \geq 5.75$

# Step 2: obtaing the cut

#### Step 2.1 

- since $x = (0,1)$ then $|S| = 1$

#### Step 2.2. obtaining the $qs$ for $x=(0,1)$

$$\min \quad −2y_1 −3y_2$$
$$\text{s. t.} \quad y_1 +2y_2 \leq \xi_1 $$
$$y1 \leq \xi_2 -1$$
$$y \in Z^+$$


- For $\xi = (2,2)$ we have:

$$\min \quad −2y_1 −3y_2$$
$$\text{s. t.} \quad y_1 +2y_2 \leq 2$$
$$y1 \leq 1$$
$$y \in Z^+$$

In [14]:
using GLPK, JuMP
m = Model(GLPK.Optimizer)
c = [-2 , -3]
A = [1 2 ; 1 0]
b = [2 ; 1]
@variable(m, 0 ≤ y[1:2], Int)
@objective(m, Min, c' * y)
@constraint(m, c1, A * y .≤ b)
optimize!(m)
println("y = ", value.(y) )
println("obj = ", objective_value(m) )

y = [0.0, 1.0]
obj = -3.0


- For $\xi = (4,3)$ we have

$$\min \quad −2y_1 −3y_2$$
$$\text{s. t.} \quad y_1 +2y_2 \leq 4$$
$$y1 \leq 2$$
$$y \in Z^+$$


In [15]:
using GLPK, JuMP
m = Model(GLPK.Optimizer)
c = [-2 , -3]
A = [1 2 ; 1 0]
b = [4 ; 2]
@variable(m, 0 ≤ y[1:2], Int)
@objective(m, Min, c' * y)
@constraint(m, c1, A * y .≤ b)
optimize!(m)
println("y = ", value.(y) )
println("obj = ", objective_value(m) )

y = [2.0, 1.0]
obj = -7.0


Therefore $qs = -5$

$$\theta \geq (qs −L)
\Big(
\sum_{i \in S} x_i -
\sum_{i \in S} x_i \Big)
- (qs - L)(|S| −1) +L$$

$$\theta \geq (-5 + 5.75)
\Big(
 x_2 -
x_1 \Big)
- (-5 + 5.75)(1 −1) - 5.75$$

$$\theta \geq 0.75 ( x_2 - x_1 ) - 5.75$$